test/transportSpec.js

```diff
@@ -0,0 +1,518 @@
+'use strict'
+
+const co = require('co')
+const chai = require('chai')
+const chaiAsPromised = require('chai-as-promised')
+chai.use(chaiAsPromised)
+const assert = chai.assert
+const expect = chai.expect
+const moment = require('moment')
+
+const ILP = require('..')
+const Transport = require('../src/lib/transport')
+const Packet = require('../src/utils/packet')
+const MockPlugin = require('./mocks/mockPlugin')
+const { wait } = require('../src/utils')
+const base64url = require('../src/utils/base64url')
+const { parsePacketAndDetails } = require('../src/utils/details')
+const Details = require('../src/utils/details')
+
+describe('Transport', function () {
+  describe('PSK', function () {
+    beforeEach(function () {
+      this.params = {
+        destinationAccount: 'test.example.alice',
+        secretSeed: Buffer.from('shh_its_a_secret', 'base64')
+      }
+    })
+
+    it('should generate psk params', function () {
+      const params = ILP.PSK.generateParams(this.params)
+
+      assert.match(params.destinationAccount, /^test\.example\.alice/)
+      assert.isString(params.sharedSecret)
+    })
+  })
+
+  describe('createPacketAndCondition', function () {
+    beforeEach(function () {
+      this.params = {
+        destinationAmount: '1',
+        destinationAccount: 'test.example.alice',
+        secret: Buffer.from('shh_its_a_secret', 'base64'),
+        data: { foo: 'bar' },
+        id: 'ee39d171-cdd5-4268-9ec8-acc349666055',
+        expiresAt: moment().format()
+      }
+
+      this.validate = (result) => {
+        const { account, amount, data } = Packet.parse(result.packet)
+
+        // the data is still encrypted, so we can't check it from just parsing
+        assert.isString(data)
+        assert.equal(amount, '1')
+        assert.match(account, /^test\.example\.alice\./)
+      }
+    })
+
+    it('should create a valid packet and condition', function () {
+      const result = Transport.createPacketAndCondition(this.params, 'psk')
+      this.validate(result)
```

Comment:
This needs to test exactly what the result looks like. Just checking that the account matches `/^test\.example\.alice\./` is insufficient because it is supposed to add to the account in a way that is important for the protocol to work properly